src/ConnectedMode.UnitTests/Suppressions/ClientSuppressionSynchronizerTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using SonarLint.VisualStudio.ConnectedMode.Suppressions;
+using SonarLint.VisualStudio.Core.Suppression;
+using SonarLint.VisualStudio.Core.Suppressions;
+using SonarLint.VisualStudio.IssueVisualization;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.TestInfrastructure;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Suppressions
+{
+    [TestClass]
+    public class ClientSuppressionSynchronizerTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<ClientSuppressionSynchronizer, IClientSuppressionSynchronizer>(
+                MefTestHelpers.CreateExport<IClientIssueStore>(),
+                MefTestHelpers.CreateExport<IIssuesFilter>());
+        }
+
+        [TestMethod]
+        public void SynchronizeIssues_IssueGetsSuppressedOnServer_LocalIssueGetUpdated()
+        {
+            var issue = CreateIssue(isSuppressedLocally: false);
+
+            var localIssues = new[] { issue };
```

Comment:
Minor: you've used `params` in the `CreateClientIssueStore` method so you don't need to explicitly create the `matches` array - you could just say:

```
  var issueFilter = CreateIssuesFilter(localIssues, issue);
```